test/autorest/generated/complexgroup/primitive.go

```diff
@@ -0,0 +1,96 @@
+package complexgroup
+
+import (
+	"context"
+	azinternal "generatortests/autorest/generated/complexgroup/internal/complexgroup"
+	"net/url"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+)
+
+// PrimitiveClient is the test Infrastructure for AutoRest Swagger
+type PrimitiveClient struct {
+	s azinternal.Service
+	u *url.URL
+	p azcore.Pipeline
+}
+
+// PrimitiveClientOptions ...
+type PrimitiveClientOptions struct {
```

Comment:
I wonder if we should have a nested `StandardClientOptions` or something that all client option structs take.  Nothing to change here just thinking out loud, we should discuss.

Reply:
Yes I was thinking the same, there are a few areas where we could probably reduce repeated code. I'll try to make a note of all of the places I think we could do this and we can discuss later.